File: typescript/typelayer-common/transportClient.ts
// Module: transportClient
import * as typelayerCommonDynamic from "./dynamic";


export class Ok {
    body: Uint8Array;
    constructor(arg: {body: Uint8Array;}) {
        Object.assign(this, arg);
    }
}

export class Uhoh {
    exceptionBody: Uint8Array;
    exceptionEncoding: typelayerCommonDynamic.Encoding;
    constructor(arg: {exceptionBody: Uint8Array; exceptionEncoding: typelayerCommonDynamic.Encoding;}) {
        Object.assign(this, arg);
    }
}

export type Response = Ok| Uhoh

export interface TransportClient {
    transportClientEncoding: () => typelayerCommonDynamic.Encoding;
    transportClientHandle: (path: Array<string>, body: Uint8Array) => Promise<Response>;
}

File: python/typelayer_common/typelayer_common/transport_client.py
from . import dynamic as typelayer_common_dynamic
from dataclasses import dataclass
from typing import List, Dict, Optional


class Response:
    pass


@dataclass
class Ok(Response):
    body: bytes

@dataclass
class Uhoh(Response):
    exception_body: bytes
    exception_encoding: typelayer_common_dynamic.Encoding

class TransportClient:
    def transport_client_encoding(self, ) -> typelayer_common_dynamic.Encoding:
        raise NotImplementedError
    def transport_client_handle(self, path: List[str], body: bytes) -> "Response":
        raise NotImplementedError